Web UI/CognitiveSearch.UI/wwwroot/js/results.js

```diff
@@ -94,19 +94,69 @@ function AddMapPoints(results) {
     }
     else {
         //Create a data source to add it to the map 
-        mapDataSource = new atlas.source.DataSource();
+        mapDataSource = new atlas.source.DataSource(null, {
+            //Tell the data source to cluster point data.
+            cluster: true,
+
+            //The radius in pixels to cluster points together.
+            clusterRadius: 45,
+
+            //The maximium zoom level in which clustering occurs.
+            //If you zoom in more than this, all points are rendered as symbols.
+            clusterMaxZoom: 15
+        });
         coordinates = UpdatePOIs(results, mapDataSource);
-        
+
         //Wait until the map resources are ready for first set up.
         resultsMap.events.add('ready', function () {
 
             //take the last coordinates.
-            if (coordinates) { resultsMap.setCamera ({ center: coordinates }); }
+            if (coordinates) { resultsMap.setCamera({ center: coordinates }); }
 
             //Add data source and create a symbol layer.
             resultsMap.sources.add(mapDataSource);
-            var symbolLayer = new atlas.layer.SymbolLayer(mapDataSource);
-            resultsMap.layers.add(symbolLayer);
+
+            //Create a bubble layer for rendering clustered data points.
+            var clusterBubbleLayer = new atlas.layer.BubbleLayer(mapDataSource, null, {
+                //Scale the size of the clustered bubble based on the number of points inthe cluster.
+                radius: [
```

Comment:
also, consider consolidating the bubble/pin symbols.. with pin being the preferred choice for showing the # of documents